rasterio/plot.py

```diff
@@ -33,38 +33,69 @@
 logger = logging.getLogger(__name__)
 
 
-def show(source, cmap='gray', with_bounds=True):
+def show(source, cmap='gray', with_bounds=True, ax=None, title=None, **kwargs):
     """Display a raster or raster band using matplotlib.
 
     Parameters
     ----------
-    source : array-like or (raster dataset, bidx)
+    source : array-like, or (raster dataset, bidx) tuple, or raster dataset, 
         If array-like, should be of format compatible with
         matplotlib.pyplot.imshow. If the tuple (raster dataset, bidx),
```

Comment:
I wonder if we should take `array-like` in raster shape and do the conversion to image shape internally.

IOW, instead of 
```
show(reshape_as_image(raster_array))
```
we'd have 
```
show(raster_array)  # conversion happens internal to the show function
```

Not sure if this would break anything, just a thought.